block-storage-attacher/watcher/pv_watcher.go

```diff
@@ -81,18 +84,40 @@ func AttachVolume(obj interface{}) {
 		lgr.Info("Persistent volume does not belong to storage class: ", zap.String("Name", pv.Name), zap.String("Storage_Class", pv.Spec.StorageClassName))
 		return
 	}
-	ModifyAttachConfig(pv)
+	attached := false
+	for !attached {
```

Comment:
1. Locking the iscsi operation will prevent multiple PV operations, which is good
2. infinite retry without a mutex lock is not good..multiple iscsi attach opeartion will start and will corrupt
3. if mutex.lock doesn't work..try file based locking(refer block-iscsi-lib)